More HTML parsing test failures in Python 3.14 beta 4

Note that 2 changes to the HTML Parser are included in Python 3.14 beta 4. ([GH-135930](https://github.com/python/cpython/pull/135930) & [GH-135664](https://github.com/python/cpython/pull/135664)). Since beta 4 was released, an additional change has been made ([GH-136908](https://github.com/python/cpython/pull/136908)). I have not tested against the unreleased change.

```
........................................../home/runner/work/markdown/markdown/markdown/core.py:436: DeprecationWarning: codecs.open() is deprecated. Use open() instead.
  output_file = codecs.open(output, "w",
......................................................................................s......................................................................................................s.s...s.ss..s....s.........s...........................................................................ss...........................................................................................................................................................FF..............................ss...........................................................................................................................................................................................................................................................FF......................................................................................................................................................................................................................................................................................
======================================================================
FAIL: test_raw_unclosed_tag_in_code_span (test_syntax.blocks.test_html_blocks.TestHTMLBlocks.test_raw_unclosed_tag_in_code_span)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/markdown/markdown/tests/test_syntax/blocks/test_html_blocks.py", line 668, in test_raw_unclosed_tag_in_code_span
    self.assertMarkdownRenders(
    ~~~~~~~~~~~~~~~~~~~~~~~~~~^
        self.dedent(
        ^^^^^^^^^^^^
    ...<15 lines>...
        )
        ^
    )
    ^
  File "/home/runner/work/markdown/markdown/markdown/test_tools.py", line 77, in assertMarkdownRenders
    self.assertMultiLineEqual(output, expected)
    ~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
AssertionError: '<p><code>&lt;div</code>.</p>\n<p><div>\nhello\n</div></p>' != '<p><code>&lt;div</code>.</p>\n<div>\nhello\n</div>'
  <p><code>&lt;div</code>.</p>
- <p><div>
? ---
+ <div>
  hello
- </div></p>
?       ----
+ </div>


======================================================================
FAIL: test_raw_unclosed_tag_in_code_span_space (test_syntax.blocks.test_html_blocks.TestHTMLBlocks.test_raw_unclosed_tag_in_code_span_space)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/markdown/markdown/tests/test_syntax/blocks/test_html_blocks.py", line 689, in test_raw_unclosed_tag_in_code_span_space
    self.assertMarkdownRenders(
    ~~~~~~~~~~~~~~~~~~~~~~~~~~^
        self.dedent(
        ^^^^^^^^^^^^
    ...<15 lines>...
        )
        ^
    )
    ^
  File "/home/runner/work/markdown/markdown/markdown/test_tools.py", line 77, in assertMarkdownRenders
    self.assertMultiLineEqual(output, expected)
    ~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
AssertionError: '<p><code>&lt;div</code>.</p>\n<p><div>\nhello\n</div></p>' != '<p><code>&lt;div</code>.</p>\n<div>\nhello\n</div>'
  <p><code>&lt;div</code>.</p>
- <p><div>
? ---
+ <div>
  hello
- </div></p>
?       ----
+ </div>


======================================================================
FAIL: test_raw_unclosed_tag_in_code_span (test_syntax.extensions.test_md_in_html.TestDefaultwMdInHTML.test_raw_unclosed_tag_in_code_span)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/markdown/markdown/tests/test_syntax/blocks/test_html_blocks.py", line 668, in test_raw_unclosed_tag_in_code_span
    self.assertMarkdownRenders(
    ~~~~~~~~~~~~~~~~~~~~~~~~~~^
        self.dedent(
        ^^^^^^^^^^^^
    ...<15 lines>...
        )
        ^
    )
    ^
  File "/home/runner/work/markdown/markdown/markdown/test_tools.py", line 77, in assertMarkdownRenders
    self.assertMultiLineEqual(output, expected)
    ~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
AssertionError: '<p><code>&lt;div</code>.</p>\n<p><div>\nhello\n</div></p>' != '<p><code>&lt;div</code>.</p>\n<div>\nhello\n</div>'
  <p><code>&lt;div</code>.</p>
- <p><div>
? ---
+ <div>
  hello
- </div></p>
?       ----
+ </div>


======================================================================
FAIL: test_raw_unclosed_tag_in_code_span_space (test_syntax.extensions.test_md_in_html.TestDefaultwMdInHTML.test_raw_unclosed_tag_in_code_span_space)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/markdown/markdown/tests/test_syntax/blocks/test_html_blocks.py", line 689, in test_raw_unclosed_tag_in_code_span_space
    self.assertMarkdownRenders(
    ~~~~~~~~~~~~~~~~~~~~~~~~~~^
        self.dedent(
        ^^^^^^^^^^^^
    ...<15 lines>...
        )
        ^
    )
    ^
  File "/home/runner/work/markdown/markdown/markdown/test_tools.py", line 77, in assertMarkdownRenders
    self.assertMultiLineEqual(output, expected)
    ~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
AssertionError: '<p><code>&lt;div</code>.</p>\n<p><div>\nhello\n</div></p>' != '<p><code>&lt;div</code>.</p>\n<div>\nhello\n</div>'
  <p><code>&lt;div</code>.</p>
- <p><div>
? ---
+ <div>
  hello
- </div></p>
?       ----
+ </div>


----------------------------------------------------------------------
Ran 1056 tests in 1.071s

FAILED (failures=4, skipped=13)
```

This failure first appeared in #1546.

--- a/docs/changelog.md
+++ b/docs/changelog.md
@@ -10,6 +10,12 @@ and this project adheres to the
 [Python Version Specification](https://packaging.python.org/en/latest/specifications/version-specifiers/).
 See the [Contributing Guide](contributing.md) for details.
 
+## [Unreleased]
+
+### Fixed
+
+* Fix handling of incomplete HTML tags in code spans in Python 3.14.
+
 ## [3.8.2] - 2025-06-19
 
 ### Fixed

--- a/markdown/htmlparser.py
+++ b/markdown/htmlparser.py
@@ -69,6 +69,20 @@ htmlparser.locatestarttagend_tolerant = re.compile(r"""
    )?
   \s*                                 # trailing whitespace
 """, re.VERBOSE)
+htmlparser.locatetagend = re.compile(r"""
+  [a-zA-Z][^`\t\n\r\f />]*           # tag name
+  [\t\n\r\f /]*                     # optional whitespace before attribute name
+  (?:(?<=['"\t\n\r\f /])[^`\t\n\r\f />][^\t\n\r\f /=>]*  # attribute name
+    (?:=                            # value indicator
+      (?:'[^']*'                    # LITA-enclosed value
+        |"[^"]*"                    # LIT-enclosed value
+        |(?!['"])[^>\t\n\r\f ]*     # bare value
+       )
+     )?
+    [\t\n\r\f /]*                   # possibly followed by a space
+   )*
+   >?
+""", re.VERBOSE)
 
 # Match a blank line at the start of a block of text (two newlines).
 # The newlines may be preceded by additional whitespace.

